packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -1248,7 +1248,11 @@ class _SearchBarState extends State<SearchBar> {
                           enabledBorder: InputBorder.none,
                           border: InputBorder.none,
                           focusedBorder: InputBorder.none,
-                          contentPadding: const EdgeInsets.symmetric(vertical: 12.0),
+                          contentPadding: EdgeInsets.zero,
+
+                          // Setting `isDense` to true to allow the text field height to be
+                          // smaller than 48.0
+                          isDense: true,
```

Comment:
Does setting isDense to true affect anything else that you don't want it to affect?  I think you might also be able to get the same effect by setting `expands` to true instead.  Maybe that will create other problems though haha.  Whatever works is fine.

Reply:
Because currently the `TextField` in `SearchBar` is only allowed to have input text and hint text, it doesn't have any other properties like perfix/prefixIcon/suffix/label/..., so I think using `isDense` should be safe:).